src/lib.rs

```diff
@@ -72,25 +72,36 @@ impl Hasher {
     /// This works just like `Hasher::new`, except that it allows for an initial
     /// CRC32 state to be passed in.
     pub fn new_with_initial(init: u32) -> Self {
-        Self::internal_new_specialized(init).unwrap_or_else(|| Self::internal_new_baseline(init))
+        Self::internal_new_specialized(init, 0)
+            .unwrap_or_else(|| Self::internal_new_baseline(init, 0))
+    }
+
+    /// Create a new `Hasher` with an initial CRC32 state.
+    ///
+    /// As `new_with_initial`, but also accepts a length (in bytes). The
+    /// resulting object can then be used with `combine` to compute `crc(a ||
+    /// b)` from `crc(a)`, `crc(b)`, and `len(b)`.
+    pub fn new_with_initial_length(init: u32, amount: u64) -> Self {
```

Comment:
Could we name this `new_with_initial_len`?

Slightly shorter and feels a bit more Rust-y (c.f. `::len` or `::set_len` methods in the stdlib).